src/spikeinterface/widgets/traces.py

```diff
@@ -111,6 +109,11 @@ def __init__(
         if channel_ids is None:
             channel_ids = rec0.channel_ids
 
+        if rec0.has_channel_location():
+            channel_locations = rec0.get_channel_locations()
+        else:
+            channel_locations = None
```

Comment:
This was a nasty bug! We wer getting the locations before sorting the channels by depth